let User,CustomErrorHandler;_c7c‍.x([["default",()=>_c7c‍.o]]);_c7c‍.w("../../models",[["User",["User"],function(v){User=v}]]);_c7c‍.w("../../services/CustomErrorHandler",[["default",["CustomErrorHandler"],function(v){CustomErrorHandler=v}]]);


const userController = {
    async me(req, res, next) {
        try {
            const user = await User.findOne({_id: req.user._id}).select('-password -updatedAt -__v');
            if(!user){
                return next(CustomErrorHandler.notFound());
            }

            res.json(user);
        } catch (error) {
            return next(error);
        }
    }
}

_c7c‍.d(userController);